web/src/pages/Main.tsx

```diff
@@ -1,9 +1,12 @@
-import { Header } from '../componenets/common/Header'
+import { Header } from '../components/common/Header'
 import { useContext } from 'react'
 import { AuthContext } from 'react-oauth2-code-pkce'
 
-export const MainPage = (): JSX.Element => {
-  const { token } = useContext(AuthContext)
+export const MainPage = (): JSX.Element | null => {
+  const { token, loginInProgress } = useContext(AuthContext)
+
+  if (loginInProgress) return null
```

Comment:
Use conditional rendering instead of returning null component

Reply:
@soofstad I reset your commit - it wasn't related to this issue (#20), so if you open a separate PR to review it, we won't have to await merging this one.